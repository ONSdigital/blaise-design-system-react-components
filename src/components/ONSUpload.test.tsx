import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { ONSUpload } from "./ONSUpload";

describe("ONS Upload Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const Props = {};

    const changeProps = {
        label: "Upload",
        description: "This is the upload",
        fileName: "file.csv",
        fileID: "file",
        accept: "yes?",
        onChange: jest.fn(),
    };

    function wrapper(renderFunc: any, props: any) {
        return renderFunc(
            <ONSUpload
                label={props.label}
                description={props.description}
                fileName={props.fileName}
                fileID={props.fileID}
                accept={props.accept}
                onChange={props.onChange}
            />,
        );
    }

    it("matches Snapshot", () => {
        expect(wrapper(render, Props)).toMatchSnapshot();
    });

    it("should handle a change", () => {
        const input = wrapper(shallow, changeProps);
        input.find("input").simulate("change", { target: { value: "abc" } });
        expect(changeProps.onChange).toHaveBeenCalled();
    });

    it.skip("should allow the user to select and upload a file", async () => {
        wrapper(render, changeProps);

        const uploadInput = screen.getByLabelText(/Upload/i);
        await userEvent.click(uploadInput);
        await userEvent.upload(uploadInput, testFile);

        expect(uploadInput.files[0]).toStrictEqual(testFile);
        expect(uploadInput.files.item(0)).toStrictEqual(testFile);
        expect(uploadInput.files).toHaveLength(1);
    });
});
