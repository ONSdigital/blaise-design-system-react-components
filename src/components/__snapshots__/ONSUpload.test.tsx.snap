// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Upload Test matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ons-field"
      >
        <p
          class="ons-field"
        >
          <label
            class="ons-label"
          >
            <br />
            <span
              class="ons-label__description"
            />
          </label>
          <input
            class="ons-input ons-input--text ons-input-type__input ons-input--upload"
            style="position: static;"
            type="file"
          />
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ons-field"
    >
      <p
        class="ons-field"
      >
        <label
          class="ons-label"
        >
          <br />
          <span
            class="ons-label__description"
          />
        </label>
        <input
          class="ons-input ons-input--text ons-input-type__input ons-input--upload"
          style="position: static;"
          type="file"
        />
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
